js/DisableUiComponent.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var $ = require('jquery');
+
+function DisableUiComponent(html) {
+    this.$html = html;
+}
+
+function preventDefaultAndStopPropagation(e) {
+    e.preventDefault();
+    e.stopPropagation();
+}
+
+DisableUiComponent.prototype.init = function () {
+    var component = this;
+
+    component.$container = this.$html.find('[data-disable-ui="true"]');
+
+    component.$container.each(function() {
+
+        var $this = $(this),
+            FORM_ELEMENTS = 'button:not(.disabled), input:not(.disabled), select:not(.disabled)',
+            LINK_ELEMENTS = 'a:not(.disabled)',
+            LABEL_ELEMENTS = 'label';
+
+        // Disable form elements
+        $this.find(FORM_ELEMENTS)
+            .on('click', preventDefaultAndStopPropagation)
+            .addClass('disabled')
+            .attr('disabled', 'disabled');
+
+        // Disable labels
+        $this.find(LABEL_ELEMENTS).addClass('u-cursor-not-allowed');
```

Comment:
I considered that. The issue being `.is-disabled` has different styles set depending on the element/class it's attached to. For example, some places it reduces color contrast, other places it adds background colors, these things resulted in lots of different opacities/styles being set on the disabled UI contents. I wanted a unified disabled UI with just the cursor and the faded opacity. It doesn't deal with the cursor either, which of course we could add. Moving to a separate specific "one use" class felt better. Thoughts?